url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/robot/SOCRobotNegotiator.java

```diff
@@ -0,0 +1,2519 @@
+        {
+            for (int pn = 0; pn < game.maxPlayers; pn++)
+            {
+                if (( ! game.isSeatVacant(pn)) &&
+                    (game.getPlayer(pn).getResources().getAmount(rsrcType) > 0))
+                {
+                    isSellingResource[pn][rsrcType] = true;
+                }
+            }
+        }
+    }
+
+    /**
+     * reset the wantsAnotherOffer array to all false
+     */
+    public void resetWantsAnotherOffer()
+    {
+        D.ebugPrintln("*** resetWantsAnotherOffer (all false) ***");
+
+        for (int rsrcType = SOCResourceConstants.CLAY;
+                rsrcType <= SOCResourceConstants.WOOD; rsrcType++)
+        {
+            for (int pn = 0; pn < game.maxPlayers; pn++)
+            {
+                wantsAnotherOffer[pn][rsrcType] = false;
+            }
+        }
+    }
+
+    /**
+     * mark a player as not selling a resource
+     *
+     * @param pn         the number of the player
+     * @param rsrcType   the type of resource
+     */
+    public void markAsNotSelling(int pn, int rsrcType)
+    {
+        D.ebugPrintln("*** markAsNotSelling pn=" + pn + " rsrcType=" + rsrcType);
+        isSellingResource[pn][rsrcType] = false;
+    }
+
+    /**
+     * mark a player as willing to sell a resource
+     *
+     * @param pn         the number of the player
+     * @param rsrcType   the type of resource
+     */
+    public void markAsSelling(int pn, int rsrcType)
+    {
+        D.ebugPrintln("*** markAsSelling pn=" + pn + " rsrcType=" + rsrcType);
+        isSellingResource[pn][rsrcType] = true;
+    }
+
+    /**
+     * mark a player as not wanting another offer
+     *
+     * @param pn         the number of the player
+     * @param rsrcType   the type of resource
+     */
+    public void markAsNotWantingAnotherOffer(int pn, int rsrcType)
+    {
+        D.ebugPrintln("*** markAsNotWantingAnotherOffer pn=" + pn + " rsrcType=" + rsrcType);
+        wantsAnotherOffer[pn][rsrcType] = false;
+    }
+
+    /**
+     * mark a player as wanting another offer
+     *
+     * @param pn         the number of the player
+     * @param rsrcType   the type of resource
+     */
+    public void markAsWantsAnotherOffer(int pn, int rsrcType)
+    {
+        D.ebugPrintln("*** markAsWantsAnotherOffer pn=" + pn + " rsrcType=" + rsrcType);
+        wantsAnotherOffer[pn][rsrcType] = true;
+    }
+
+    /**
+     * @return true if the player is marked as wanting a better offer
+     *
+     * @param pn         the number of the player
+     * @param rsrcType   the type of resource
+     */
+    public boolean wantsAnotherOffer(int pn, int rsrcType)
+    {
+        return wantsAnotherOffer[pn][rsrcType];
+    }
+
+    /***
+     * make an offer to another player
+     *
+     * @param targetPiece  the piece that we want to build
+     * @return the offer we want to make, or null for no offer
+     */
+    public SOCTradeOffer makeOffer(SOCPossiblePiece targetPiece)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 89. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 343-357 into a separate method.